t/ui/15-comments.t

```diff
@@ -474,6 +474,50 @@ subtest 'editing when logged in as regular user' => sub {
         is(scalar @{$driver->find_elements('.comment-row')}, 1, 'only 1 comment present on last page');
       }
       for (@group_overview_urls);
+
+    subtest 'commenting on group overview' => sub {
+        my $group_url = '/group_overview/1001';
+        my @need_deleted_comments;
+        my $comments = $schema->resultset('Comments');
+
+        for (my $i = 0; $i < 100; $i++) {
+            my $comment = $comments->create(
+                {
+                    group_id => 1001,
+                    text     => 'example ' . $i,
+                    user_id  => '99901'
+                });
+
+            push @need_deleted_comments, $comment;
+        }
+
+        $driver->get($group_url . '?comments_limit=110');
+        my @pagination_buttons = $driver->find_elements('.comments-pagination a', 'css');
+        is(scalar @pagination_buttons,                       1,   'Only one button present');
+        is(scalar @{$driver->find_elements('.comment-row')}, 105, '105 comments present');
+
+        $driver->get($group_url . '?comments_limit=2');
+        @pagination_buttons = $driver->find_elements('.comments-pagination a', 'css');
+        is(scalar @pagination_buttons,
+            54, '54 pagination buttons present (one is >>)' . $group_url . '?comments_limit=2');
+        is(scalar @{$driver->find_elements('.comment-row')}, 2, 'only 2 comments present');
+
+        # A page in the middle (page 25) shows 2 comments and to check all buttons are showed
+        $pagination_buttons[25]->click();
+        @pagination_buttons = $driver->find_elements('.comments-pagination a', 'css');
+        is(scalar @pagination_buttons,
+            55, '54 pagination buttons present (Including << and >>)' . $group_url . '?comments_limit=2');
+        is(scalar @{$driver->find_elements('.comment-row')}, 2, 'only 2 comments present');
+
+        # Last page has only one comment
+        $pagination_buttons[53]->click();
+        @pagination_buttons = $driver->find_elements('.comments-pagination a', 'css');
+        is(scalar @pagination_buttons,
+            54, '54 pagination buttons present (one is <<)' . $group_url . '?comments_limit=2');
+        is(scalar @{$driver->find_elements('.comment-row')}, 1, 'only 1 comment present on last page');
+
+        $_->delete for @need_deleted_comments;
+    }
 };
```

Comment:
wow, that is certainly impressive. But honestly I think this is rather longish and probably to detailed and too specific to be easy to maintain. We already have a block just above that tests  that pagination is present and in here you can also not assert that "it looks nice" on the screen – that's still the issue you want to solve, right? – so I might have missed this point. But do you need to add any test at all? You are not adding new features but basically reworking the existing functionality so to my understanding you do not need to add any tests at all. Or what was missing from test coverage in before?

Reply:
I'm a little bit torn whether I like this test or not.

> and too specific to be easy to maintain

It doesn't seem *that* bad to maintain but I'm wondering about the performance of the test. And maybe one can check the same code paths without that many comments by setting `comments_limit` to something low (which some of part of the test already does).

> We already have a block just above that tests that pagination is present [...] Or what was missing from test coverage in before?

The existing tests failed to reveal that the page parameter is possibly added multiple times to the URL query. By the way, has this problem been tackled yet?

Reply:
The test I added is to ensure that is painting all the pages from 1 to the last page. This validate the fix of this PR. I think it should be tested in this way. 
The above test is using the url parent_group_overview/1 I am using /group_overview/1001 to allow the creation of a enough pages to ensure that the fix is working. 
Both tests could be merged but integrating some checks from the previous test into the new one.

Reply:
Considering equivalent ranges for testing I do not understand what should be the benefit to test every single page if this is what you mean. I appreciate that you want to extend testing but sorry I still do not understand what's the additional value.

Reply:
In the last commit I removed this subtest